tests/glhe/borehole/test_base.py

```diff
@@ -1,10 +1,15 @@
 import unittest
 
 from glhe.borehole.base import BoreholeBase
+from glhe.borehole.simpleHX import BoreholeSimpleHX
 
 
 class TestBoreholeSimpleHX(unittest.TestCase):
 
+    def test_abstract_instantiation_fails(self):
+        with self.assertRaises(Exception):
```

Comment:
I'm not getting an exception raised on my machine (or on Travis). Do I need to add something to the base class __init__?

Reply:
Well I'm not sure.  I had assumed that if you had a class and decorated a function with the abstract decorator, then if you tried to instantiate a raw version of that class, it would fail.  But maybe that's not true.  There are definitely a few anomalies you have to account for when trying to maintain a codebase for both Python 2 and 3, and this may just be one of them.  I think it would be worth first trying to have Travis run with both 2 and 3, and seeing what Travis sees.  If it throws an exception on 2, but doesn't on 3, then we can use the [six](https://pythonhosted.org/six/) Python library to check for which version is in place and do a few specialties here and there.